### ISSUE TITLE
CgovCoreTools is incorrectly referencing the Exception class

### ISSUE THREAD

**Issue by bryanpizzillo** [NCIOCPL/cgov-digital-platform#3089]
### Issue description
The TL;DR; is that we are not throwing the `Exception` class correctly. We should `throw new \Exception(...)` instead of `throw new Exception(...)`. 

Now, how we got this case to trigger, and how we did not have tests that showed us how bad this is, are the real issues. Basically, we had a number of instances of the following code get called, where for some reason the passed in `$entity` did not have a `field_media_link` field. This should be impossible, in fact, it is only possible if there was a major content change OR, the configuration was misread.
 
```
if ($bundle === 'cgov_media_link' && $operation === 'view') {
    return \Drupal::service('cgov_core.tools')->filterAccessForDependantEntity($entity, 'field_media_link');
 }
```

The problematic code is in filterAccessForDependantEntity, which does the check. So this should be updated to throw a `\Exception` and it should get a unit test. (There is not too much to mock...)

> **ESTIMATE** TBD

### Steps to reproduce the issue
Not really possible.

### What's the expected result?
That the error is correctly thrown. That a unit test for the case exists.

### Additional details / screenshot
Error Stack
```
Error: Uncaught exception 'Error' with message 'Class 'Drupal\cgov_core\Exception' not found' in /mnt/www/html/ncigov01live/docroot/profiles/custom/cgov_site/modules/custom/cgov_core/src/CgovCoreTools.php:490
in Drupal\cgov_core\CgovCoreTools::filterAccessForDependantEntity called at /mnt/www/html/ncigov01live/docroot/profiles/custom/cgov_site/modules/custom/cgov_list/cgov_list.module (32)
in cgov_list_paragraph_access called at ? (?)
in call_user_func_array called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Extension/ModuleHandler.php (403)
in Drupal\Core\Extension\ModuleHandler::invokeAll called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Entity/EntityAccessControlHandler.php (97)
in Drupal\Core\Entity\EntityAccessControlHandler::access called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Entity/ContentEntityBase.php (709)
in Drupal\Core\Entity\ContentEntityBase::access called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Field/Plugin/Field/FieldFormatter/EntityReferenceFormatterBase.php (184)
in Drupal\Core\Field\Plugin\Field\FieldFormatter\EntityReferenceFormatterBase::checkAccess called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Field/Plugin/Field/FieldFormatter/EntityReferenceFormatterBase.php (55)
in Drupal\Core\Field\Plugin\Field\FieldFormatter\EntityReferenceFormatterBase::getEntitiesToView called at /mnt/www/html/ncigov01live/docroot/modules/contrib/entity_reference_revisions/src/Plugin/Field/FieldFormatter/EntityReferenceRevisionsEntityFormatter.php (131)
in Drupal\entity_reference_revisions\Plugin\Field\FieldFormatter\EntityReferenceRevisionsEntityFormatter::viewElements called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Field/FormatterBase.php (89)
in Drupal\Core\Field\FormatterBase::view called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Field/Plugin/Field/FieldFormatter/EntityReferenceFormatterBase.php (76)
in Drupal\Core\Field\Plugin\Field\FieldFormatter\EntityReferenceFormatterBase::view called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Entity/Entity/EntityViewDisplay.php (263)
in Drupal\Core\Entity\Entity\EntityViewDisplay::buildMultiple called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Entity/EntityViewBuilder.php (351)
in Drupal\Core\Entity\EntityViewBuilder::buildComponents called at /mnt/www/html/ncigov01live/docroot/core/modules/node/src/NodeViewBuilder.php (24)
in Drupal\node\NodeViewBuilder::buildComponents called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Entity/EntityViewBuilder.php (293)
in Drupal\Core\Entity\EntityViewBuilder::buildMultiple called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Entity/EntityViewBuilder.php (250)
in Drupal\Core\Entity\EntityViewBuilder::build called at ? (?)
in call_user_func_array called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Security/DoTrustedCallbackTrait.php (100)
in Drupal\Core\Render\Renderer::doTrustedCallback called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Render/Renderer.php (781)
in Drupal\Core\Render\Renderer::doCallback called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Render/Renderer.php (372)
in Drupal\Core\Render\Renderer::doRender called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Render/Renderer.php (200)
in Drupal\Core\Render\Renderer::render called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Render/MainContent/HtmlRenderer.php (226)
in Drupal\Core\Render\MainContent\HtmlRenderer::Drupal\Core\Render\MainContent\{closure} called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Render/Renderer.php (573)
in Drupal\Core\Render\Renderer::executeInRenderContext called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Render/MainContent/HtmlRenderer.php (227)
in Drupal\Core\Render\MainContent\HtmlRenderer::prepare called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/Render/MainContent/HtmlRenderer.php (117)
in Drupal\Core\Render\MainContent\HtmlRenderer::renderResponse called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/EventSubscriber/MainContentViewSubscriber.php (90)
in Drupal\Core\EventSubscriber\MainContentViewSubscriber::onViewRenderArray called at ? (?)
in call_user_func called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Component/EventDispatcher/ContainerAwareEventDispatcher.php (111)
in Drupal\Component\EventDispatcher\ContainerAwareEventDispatcher::dispatch called at /mnt/www/html/ncigov01live/vendor/symfony/http-kernel/HttpKernel.php (156)
in Symfony\Component\HttpKernel\HttpKernel::handleRaw called at /mnt/www/html/ncigov01live/vendor/symfony/http-kernel/HttpKernel.php (68)
in Symfony\Component\HttpKernel\HttpKernel::handle called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/StackMiddleware/Session.php (57)
in Drupal\Core\StackMiddleware\Session::handle called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/StackMiddleware/KernelPreHandle.php (47)
in Drupal\Core\StackMiddleware\KernelPreHandle::handle called at /mnt/www/html/ncigov01live/docroot/core/modules/page_cache/src/StackMiddleware/PageCache.php (191)
in Drupal\page_cache\StackMiddleware\PageCache::fetch called at /mnt/www/html/ncigov01live/docroot/core/modules/page_cache/src/StackMiddleware/PageCache.php (128)
in Drupal\page_cache\StackMiddleware\PageCache::lookup called at /mnt/www/html/ncigov01live/docroot/core/modules/page_cache/src/StackMiddleware/PageCache.php (82)
in Drupal\page_cache\StackMiddleware\PageCache::handle called at /mnt/www/html/ncigov01live/docroot/modules/contrib/shield/src/ShieldMiddleware.php (91)
in Drupal\shield\ShieldMiddleware::handle called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/StackMiddleware/ReverseProxyMiddleware.php (47)
in Drupal\Core\StackMiddleware\ReverseProxyMiddleware::handle called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/StackMiddleware/NegotiationMiddleware.php (52)
in Drupal\Core\StackMiddleware\NegotiationMiddleware::handle called at /mnt/www/html/ncigov01live/vendor/stack/builder/src/Stack/StackedHttpKernel.php (23)
in Stack\StackedHttpKernel::handle called at /mnt/www/html/ncigov01live/docroot/core/lib/Drupal/Core/DrupalKernel.php (708)
in Drupal\Core\DrupalKernel::handle called at /mnt/www/html/ncigov01live/docroot/index.php (19)
```

**Comment by blairlearn** [comment]
Reopening as the fix is incomplete. In addition to fixing the syntax error, unit tests were also supposed to be added.

(At a glance, this is very similar to #3088, however that issue didn't require additional changes.)

**Comment by blairlearn** [comment]
The original defect is fixed.  The missing unit tests will be addressed in #4692 and #4514 as approprirate.
